src/System.Net.Http/src/System/Net/Http/HPack/StaticTable.cs

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Text;
+
+namespace System.Net.Http.HPack
+{
+    internal class StaticTable
+    {
+        private static readonly StaticTable _instance = new StaticTable();
+
+        private readonly Dictionary<int, int> _statusIndex = new Dictionary<int, int>
+        {
+            [200] = 8,
+            [204] = 9,
+            [206] = 10,
+            [304] = 11,
+            [400] = 12,
+            [404] = 13,
+            [500] = 14,
+        };
+
+        private StaticTable()
+        {
+        }
```

Comment:
Why is this designed as a singleton rather than just having static methods?  Is there a perf advantage to that, e.g. caching the instance into a local and then having instance method calls / accessing instance state be faster than accessing static state?

Reply:
No idea. I doubt it was a considered decision. Do you have a strong opinion?

Reply:
> Do you have a strong opinion?

I'd prefer statics unless there's a good reason for it to be a singleton.

Reply:
Ok, fixed